junit-jupiter-api/src/main/java/org/junit/jupiter/api/parallel/ResourceLocksProvider.java

```diff
@@ -75,16 +76,18 @@ default Set<Lock> provideForClass(Class<?> testClass) {
 	 * the same semantics as annotating a nested test class with an analogous
 	 * {@code @ResourceLock(value, mode)} declaration.
 	 *
+	 * @param enclosingInstanceTypes the runtime types of the enclosing
+	 * instances; never {@code null}
 	 * @param testClass a nested test class for which to add shared resources
```

Comment:
As in #4266...

We need to document the order of the enclosing instance types list (top-down, bottom-up, whether the list includes/excludes the _current_ nested class, etc.), either in every affected method or at the class level (and then reference the class-level documentation).

Reply:
Done in b13ac6e0bcde90585b8e68f2b6be569bd5b3b149